src/disclosure-policy/verifyAuthorizationForOpenid4VpAuthorizationRequest.ts

```diff
@@ -0,0 +1,101 @@
+import { type AgentContext, type DcqlCredentialsForRequest, X509Certificate } from '@credo-ts/core'
+import type { OpenId4VpAuthorizationRequestPayload, OpenId4VpResolvedAuthorizationRequest } from '@credo-ts/openid4vc'
+import { type AllowListPolicy, validateAllowListPolicy } from './validateAllowlistPolicy'
+import {
+  type AttributeBasedAccessControlPolicy,
+  validateAttributeBasedAccessControlPolicy,
+} from './validateAttributeBasedAccessControlPolicy'
+import { type RootOfTrustPolicy, validateRootOfTrustPolicy } from './validateRootOfTrustPolicy'
+import { isAuthorizationAttestation } from './verifyAuthorizationAttestation'
+
+export type AuthzPolicy = AllowListPolicy | RootOfTrustPolicy | AttributeBasedAccessControlPolicy
```

Comment:
We generally don't use abbreviations

```suggestion
export type AuthorizationPolicy = AllowListPolicy | RootOfTrustPolicy | AttributeBasedAccessControlPolicy
```